x/wasm/types/authz.go

```diff
@@ -39,16 +43,23 @@ func (a StoreCodeAuthorization) MsgTypeURL() string {
 
 // Accept implements Authorization.Accept.
 func (a *StoreCodeAuthorization) Accept(ctx sdk.Context, msg sdk.Msg) (authztypes.AcceptResponse, error) {
-	var code []byte
-	var permission AccessConfig
-	switch msg := msg.(type) {
-	case *MsgStoreCode:
-		code = msg.WASMByteCode
-		permission = *msg.InstantiatePermission
-	default:
+	storeMsg, ok := msg.(*MsgStoreCode)
+	if !ok {
 		return authztypes.AcceptResponse{}, sdkerrors.ErrInvalidRequest.Wrap("unknown msg type")
 	}
 
+	code := storeMsg.WASMByteCode
+	permission := *storeMsg.InstantiatePermission
+
+	if ioutils.IsGzip(code) {
+		// TODO: consume gas
+		wasmCode, err := ioutils.Uncompress(code, int64(MaxWasmSize))
+		if err != nil {
+			return authztypes.AcceptResponse{}, sdkerrors.ErrInvalidRequest.Wrap(errorsmod.Wrap(err, "uncompress wasm archive").Error())
```

Comment:
This line would work with a Go stdlib error without a change. The error message is wrapped into a `ErrInvalidRequest` type. The registered `errLimit` namespace/code is ignored anyway.